File: src/main/java/no/sanchezrolfsen/framework/selenium/config/SeleniumOptions.java
package no.sanchezrolfsen.framework.selenium.config;

import static java.lang.String.format;
import static no.sanchezrolfsen.framework.selenium.config.BrowserType.*;

public class SeleniumOptions {
    public static BrowserType stringToBrowserType(String browserType) throws IllegalArgumentException {
        return switch (browserType.toUpperCase()) {
            case "CHROME" -> CHROME;
            case "CHROME_HEADLESS" -> CHROME_HEADLESS;
            case "FIREFOX" -> FIREFOX;
            case "FIREFOX_HEADLESS" -> FIREFOX_HEADLESS;
            default -> throw new IllegalArgumentException(format("Browser '%s' is not supported in the framework", browserType));
        };
    }
}

File: src/main/kotlin/no/sanchezrolfsen/framework/selenium/config/SeleniumOptions.kt
package no.sanchezrolfsen.framework.selenium.config

internal object SeleniumOptions {

    @JvmStatic
    fun stringToBrowserType(browserType: String): BrowserType =
        when (browserType.uppercase()) {
            "CHROME" -> BrowserType.CHROME
            "CHROME_HEADLESS" -> BrowserType.CHROME_HEADLESS
            "FIREFOX" -> BrowserType.FIREFOX
            "FIREFOX_HEADLESS" -> BrowserType.FIREFOX_HEADLESS
            else -> throw IllegalArgumentException("Browser '$browserType' is not supported in the framework")
        }
}
